commons/src/main/java/org/infinispan/commons/api/functional/Param.java

```diff
@@ -44,6 +44,29 @@
    P get();
 
    /**
+    * Should this paramater go over the wire?
+    */
+   SerializeMethod serializeMethod();
```

Comment:
Cross posting what I said in #4303. My feeling is that overall this method and enum pollute the interface itself when there's some clear ways to avoid the issue without the need to add extra methods.